processor/eventfilter/eventfilter.go

```diff
@@ -69,3 +77,325 @@ func GetSupportedMessageEvents(destination *backendconfig.DestinationT) ([]strin
+	case []T:
+		return value
+	case []interface{}:
+		result := make([]T, len(value))
+		for i, v := range value {
+			var ok bool
+			result[i], ok = v.(T)
+			if !ok {
+				return []T{}
+			}
+		}
+		return result
+	}
+	return []T{}
+}
+
+type EventProperty[T EventPropsTypes] struct {
+	AllowAll    bool
+	AllowedVals []T
+}
+
+/*
+Converts the eventPropertyMap to a struct
+In an eventPropertyMap we expect only one of [allowAll, allowedValues] properties
+
+Possible cases
+
+ 1. { [eventProperty]: { allowAll: true/false }  }
+
+    - One of the expected ways
+
+    - EventProperty{AllowAll: true/false}
+
+ 2. { [eventProperty]: { allowedValues: [val1, val2, val3] }  }
+
+    - One of the expected ways
+
+    - Output: EventProperty{AllowedValues: [val1, val2, val3]}
+
+ 3. { [eventProperty]: { allowedValues: [val1, val2, val3], allowAll: true/false }  }
+
+    - Not expected
+
+    - EventProperty{AllowAll: true/false}
+
+ 4. { [eventProperty]: { }  }
+
+    - Not expected
+
+    - EventProperty{AllowAll:false}
+*/
+func ConvertEventPropMapToStruct[T EventPropsTypes](eventPropMap map[string]interface{}) *EventProperty[T] {
+	var eventPropertyStruct EventProperty[T]
+	for _, key := range []string{"allowAll", "allowedValues"} {
+		val, ok := eventPropMap[key]
+		if !ok {
+			pkgLogger.Debugf("'%v' not found in eventPropertiesMap(supportedConnectionModes.sourceType.connectionMode.[eventProperty])", key)
+			continue
+		}
+		switch key {
+		case "allowAll":
+			allowAll, convertable := val.(bool)
+			if !convertable {
+				return nil
+			}
+			eventPropertyStruct.AllowAll = allowAll
+		case "allowedValues":
+			allowedVals := ConvertToArrayOfType[T](val)
+			eventPropertyStruct.AllowedVals = allowedVals
+		}
+	}
+	return &eventPropertyStruct
+}
+
+type supportedEventProperties struct {
+	EventPropsMap  map[string]interface{}
+	Allow          bool
+	FailedResponse *transformer.TransformerResponseT
+}
+
+/*
+We obtain the supported event properties map data from destination.Definition.Config.supportedConnectionModes
+Structure looks like this
+
+	[sourceType]: {
+		[connectionMode]: {
+			[eventProperty]: [...supportedEventPropertyValues]
+		}
+	}
+
+@returns
+
+	{
+		[eventProperty]: [...supportedEventPropertyValues]
+	}
+*/
+func getSupportedEventPropertiesMap(destinationDefinitionConfig map[string]interface{}, srcType, destConnectionMode string, defaultBehaviour bool) supportedEventProperties {
+	supportedConnectionModesI, connModesOk := destinationDefinitionConfig["supportedConnectionModes"]
+	if !connModesOk {
+		pkgLogger.Debug("supportedConnectionModes not present, filtering event based on default behaviour")
+		return supportedEventProperties{Allow: defaultBehaviour}
+	}
+	supportedConnectionModes := supportedConnectionModesI.(map[string]interface{})
+
+	supportedEventPropsMapI, supportedEventPropsLookupErr := misc.NestedMapLookup(supportedConnectionModes, srcType, destConnectionMode)
+	if supportedEventPropsLookupErr != nil {
```

Comment:
Looks bit complicated, why don't be look for srcType first and take some decisions based on that then check for destConnectionMode

Reply:
We can do the checking 1-by-1 it looked rather un-comfortable.

Reply:
but level can lead to some errors and not sure if this is making code more readable